BlockSettleHW/ledger/ledgerDevice.cpp

```diff
@@ -67,19 +68,30 @@ namespace {
 
    uint16_t receiveApduResult(hid_device* dongle, QByteArray& response) {
       response.clear();
-      uint16_t chunkNumber = 0;
+      uint16_t expectedChunkIndex = 0;
 
       unsigned char buf[Ledger::CHUNK_MAX_BLOCK];
-      uint16_t result = hid_read(dongle, buf, Ledger::CHUNK_MAX_BLOCK);
+      int result = hid_read(dongle, buf, Ledger::CHUNK_MAX_BLOCK);
       if (result < 0) {
          return result;
       }
 
       std::vector<uint8_t> buff(&buf[0], &buf[0] + 64);
 
       QByteArray chunk(reinterpret_cast<char*>(buf), Ledger::CHUNK_MAX_BLOCK);
-      assert(chunkNumber++ == chunk.mid(3, 2).toHex().toInt());
+      auto checkChunkIndex = [&]() {
```

Comment:
Capturing everything is a bad practice - capture only used variables in a lambda.

Reply:
Fixed, but I guess this rule shouldn't be covered case for a local helpful lambda. Anyway fixed